### PR TITLE
fix(client): eyes not blinking in english animations

### PR DESCRIPTION
**Pull request by moT01**
The eyes in the English animations aren't blinking sometimes. I found that the browser is cancelling the requests to load the images:

Screenshot 2024-02-05 at 11 13 26 AM

Sometimes some of the images do seem to load. Perhaps some kind of issue with the blinking timeout in the component, causing the browser to cancel the request. The mouth images never seem to have this issue, which use a similar timeout. Not sure if this has always been an issue - pretty sure it used to work just fine. Anyway, this PR preloads all the images in the scene, which seems to fix the issue. If anyone has an idea for an alternative fix, feel free to share.

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
I wasn't able to reproduce the bug. Chrome, Brave and Firefox all seem fine and all look like this:
Blinky

**Comment by moT01** [comment]
Hmm, perhaps we needed an issue with some more discussion. Daniel was able to reproduce it and I am able to consistently reproduce it. Not sure exactly why it's happening though.

**Comment by ojeytonwilliams** [comment]
Could you link to the discussion where you see it? Also, is it consistently messed up across different browsers?

**Comment by moT01** [comment]
There is no discussion. Daniel messaged me on chat and I just looked into it. I was able to reproduce it on chrome and firefox - so it seemed like it was probably pretty prevalent.

**Comment by ojeytonwilliams** [comment]
Sorry, I thought you meant a particular dialogue made it obvious.

Anyways, I've been staring at lots of dialogues and they're all blinking, so I'm not sure how to reproduce this.

**Comment by ojeytonwilliams** [comment]
That's right, Huyen. I've only tested this on Linux. Just to confirm: you don't see the issue on this PR?

**Comment by zairahira** [comment]
I was able to reproduce the issue in Linux(Chrome + Firefox). The eyes weren't blinking at all.

I have tested Tom's changes and now I can see the eyes are blinking:

[Screencast from 19-02-24 21:41:57.webm](https://github.com/freeCodeCamp/freeCodeCamp/assets/33151350/d2980206-d2d8-4fbc-99e8-1dcc0818b5c2)

**Review by naomi-lgbt — APPROVED** [review]

**Review by huyenltnguyen — APPROVED** [review]
Approving, but I'll let @ojeytonwilliams give the final signoff 🙂

File: client/src/templates/Challenges/components/scene/scene.tsx [hunk 1]
```diff
@@ -21,11 +21,32 @@ export function Scene({ scene }: { scene: FullScene }): JSX.Element {
     new Audio(`${sounds}/${audio.filename}${audioTimestamp}`)
   );
 
+  const loadImage = (src: string) => {
+    const img = new Image();
+    img.src = src;
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/components/scene/scene.tsx:26]:
```suggestion
  const loadImage = (src: string | null) => {
    if (src) new Image().src;
```
It's a little cleaner and lets us tidy up below.

File: client/src/templates/Challenges/components/scene/scene.tsx [hunk 2]
```diff
@@ -21,11 +21,32 @@ export function Scene({ scene }: { scene: FullScene }): JSX.Element {
...
+    loadImage(`${backgrounds}/${setup.background}`);
+
+    setup.characters.forEach(character => {
+      const characterImages = characterAssets[character.character];
+      Object.values(characterImages).forEach(src => {
+        if (src) {
+          loadImage(src);
+        }
+      });
+    });
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/components/scene/scene.tsx:44]:
```suggestion
    setup.characters
      .map(({ character }) => Object.values(characterAssets[character]))
      .flat()
      .forEach(loadImage);
```
The linear pipeline is easier to read, imo.

File: client/src/templates/Challenges/components/scene/scene.tsx [hunk 3]
```diff
@@ -21,11 +21,32 @@ export function Scene({ scene }: { scene: FullScene }): JSX.Element {
...
+
+    commands.forEach(command => {
+      if (command.background) loadImage(`${backgrounds}/${command.background}`);
+    });
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/components/scene/scene.tsx:48]:
```suggestion
    commands
      .map(({ background }) =>
        background ? `${backgrounds}/${background}` : null
      )
      .forEach(loadImage);
```
Again, this simplifies the logical flow.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Thanks for confirming the fix works (and the explanation, Huyen). I'm happy with the fix, but I think it could be cleaned up a little.

**Review by ojeytonwilliams — APPROVED** [review]
Thanks, Tom. This LGTM :+1:

**Pull request merged**